csrc/multi_tensor_adam.cu

```diff
@@ -189,11 +300,11 @@ struct AdamCapturableFunctor
         int i = i_start + threadIdx.x + ii*blockDim.x;
         if(i < n && i < chunk_size)
         {
-	  g[i] = g[i] * (*inv_scale);
-          r_g[ii] = g[i];
-          r_p[ii] = p[i];
-          r_m[ii] = m[i];
-          r_v[ii] = v[i];
+          r_g[ii] = static_cast<MATH_T>(g[i]) * (*inv_scale);
+          g[i] = static_cast<T>(r_g[ii]); // Do we need this write?
```

Comment:
grads should be updated unless `inv_scale == 1.0`

rel: https://github.com/NVIDIA/apex/pull/1560#discussion_r1087915084